google/generativeai/types/retriever_types.py

```diff
@@ -203,32 +201,24 @@ def create_document(
         if client is None:
             client = get_default_retriever_client()
 
-        if not name and not display_name:
-            raise ValueError("Either the document name or display name must be specified.")
-
         document = None
-        if name:
-            if re.match(_DOCUMENT_NAME_REGEX, name):
-                document = glm.Document(
-                    name=name, display_name=display_name, custom_metadata=custom_metadata
-                )
-            elif f"corpora/{self.name}/documents/" not in name:
-                document_name = f"{self.name}/documents/" + re.sub(_PATTERN, "", name)
-                document = glm.Document(
-                    name=document_name, display_name=display_name, custom_metadata=custom_metadata
-                )
-            else:
-                raise ValueError(
-                    f"Document name must be formatted as {self.name}/document/<document_name>."
-                )
+        if valid_name(name):
+            document_name = f"{self.name}/documents/{name}"
+            document = glm.Document(
+                name=document_name, display_name=display_name, custom_metadata=custom_metadata
+            )
+        else:
+            raise ValueError(
+                f"`name` parameter contains non-alphanumeric characters or is longer than 40 characters. Enter an alphanumeric name which can contain dashes that is less than 40 characters, but the name must not begin or end with a dash for the `Document`. The name entered will be formatted as {self.name}/documents/<document_name>."
```

Comment:
This is used almost verbatim in several places. 

Consider just defining a constant in retreiver_types.

Reply:
I think I can make part of the statement a constant, but I do reference the name of the object here -- is there a way to also cleanly include that in the constant?